### ISSUE TITLE
Output prediction probability in classification

### ISSUE THREAD

**Issue by daikikatsuragawa** [pycaret/pycaret#294]
I want PyCaret to output not only labels at the time of classification using LightGBM or logistic regression, but also their predictive probabilities. I propose extensions of such functions.

**Comment by Virajdatt** [comment]
Do you mean while training or during inference?

**Comment by daikikatsuragawa** [comment]
I want PyCaret to output it during inference.

**Comment by Virajdatt** [comment]
The create_model() method returns the model from sklearn, xgboost etc...
and each of these have the predict_proba method, we can use it as following

LGBM = create_model("lightgbm")

LGBM.predict_proba(test_data)

Hope this helps.

<img width="510" alt="Screen Shot 2020-07-19 at 9 05 56 PM" src="https://user-images.githubusercontent.com/20900791/87880063-3f619700-ca0c-11ea-9bb0-138dcfed3328.png">

**Comment by pycaret** [comment]
@daikikatsuragawa 

When you use predict_model for inference on new dataset. Two new columns are attached to the dataframe returned. "Label" and "Score". Where Label is the actual target and "Score" is the probability of positive class.

Alternatively, what @Virajdatt suggested is also right. Incase you want to create your own inference, you can use predict_proba method on estimator.

Hope this helps.

**Comment by pycaret** [comment]
@daikikatsuragawa See this example:

![image](https://user-images.githubusercontent.com/58118658/87996619-ca9c6480-cac0-11ea-9920-b2f8fcbc026a.png)

If this helps, please close the issue.

**Comment by nurrochmanmuhammad** [comment]
what about prep_pipe of features, is it already conditioned using that way?

**Comment by AgastyaData** [comment]
> @daikikatsuragawa
> 
> When you use predict_model for inference on new dataset. Two new columns are attached to the dataframe returned. "Label" and "Score". Where Label is the actual target and "Score" is the probability of positive class.
> 
> Alternatively, what @Virajdatt suggested is also right. Incase you want to create your own inference, you can use predict_proba method on estimator.
> 
> Hope this helps.

Probability for positive class? or probability of predicted class?

**Comment by AgastyaData** [comment]
> @daikikatsuragawa See this example:
> 
> ![image](https://user-images.githubusercontent.com/58118658/87996619-ca9c6480-cac0-11ea-9920-b2f8fcbc026a.png)
> 
> If this helps, please close the issue.

<img width="1002" alt="Screenshot 2021-03-24 at 1 21 13 PM" src="https://user-images.githubusercontent.com/80875862/112274077-13d23300-8ca4-11eb-98b2-ea1acab02a00.png">

**Comment by bimhud** [comment]
I have exactly the same issue about the "Score" column. You can see the distribution of the score for the "positive (1)" and negative (0) class.

Regardless of the target label, the score for any class are greater than .5

![image](https://user-images.githubusercontent.com/27791562/113526301-e9b23680-95fc-11eb-8fcf-f2a1102a3dd4.png)

![image](https://user-images.githubusercontent.com/27791562/113526350-149c8a80-95fd-11eb-9fbd-e4634bc7334e.png)

Anyone has any idea about this?

**Comment by bimhud** [comment]
@AgastyaData  Found the issue, the "score" is considered  as probability of predicted class. probability of predicted class. It can seen clearly by turn on the raw_data from predict_model such as

predict_model(top1, data=d_test,raw_score=True)

There will be two score for positive and negative classes. This has explained the issue I've got. hope yours as well.

![image](https://user-images.githubusercontent.com/27791562/113528051-ce4a2a00-9602-11eb-9d72-59b24e6b24c3.png)
